Detectors/TRD/reconstruction/src/CruRawReader.cxx

```diff
@@ -523,29 +678,40 @@ int CruRawReader::processHalfCRU(int iteration)
         LOGF(info, "Half-CRU link %i has zero link size", currentlinkindex);
       }
     }
-
     if (linkstart != linkend) { // if link is not empty
       auto trackletparsingstart = std::chrono::high_resolution_clock::now();
-      if (mHeaderVerbose) {
+      if (mOptions[TRDVerboseErrorsBit]) {
         LOGF(info, "Tracklet parser starting at offset %u and processing up to %u words", mHBFoffset32, currentlinksize32);
       }
-
       // for now we are using 0 i.e. from rdh FIXME figure out which is authoritative between rdh and ori tracklethcheader if we have it enabled.
       int trackletWordsRead = mTrackletsParser.Parse(&mHBFPayload, linkstart, linkend, mFEEID, side, detectorId, stack, layer, mCurrentEvent, &mEventRecords, mOptions, mTrackletHCHeaderState); // this will read up to the tracklet end marker.
+      if (mTrackletsParser.dumpLink()) {
+        //dump the link that cause the error
+        // the call to dumpLink resets the boolean to false;
```

Comment:
but with every call to `incParsingError()` the boolean is set to true again. That way you would dump the data all the time and not only once upon error per run. Or was it intended like that?

Reply:
Intended like that. In the initial comments for this pr, I will implement a counter per error, but want this through now. Can fix that later.